iroh-p2p/src/service.rs

```diff
@@ -472,12 +490,31 @@ pub async fn build_transport(local_key: Keypair) -> Boxed<(PeerId, StreamMuxerBo
         core::upgrade::SelectUpgrade::new(yamux_config, mplex_config)
     };
 
-    transport
-        .upgrade(core::upgrade::Version::V1Lazy)
-        .authenticate(auth_config)
-        .multiplex(mplex_config)
-        .timeout(Duration::from_secs(20)) // TODO: configurable
-        .boxed()
+    if config.relay_client {
+        let (relay_transport, relay_client) =
+            libp2p::relay::v2::client::Client::new_transport_and_behaviour(
```

Comment:
Hope the creation of the `NetworkBehaviour` and the `Transport` was OK to follow. Tried to make it impossible to missuse at compile time, likely at the expense of being easy to use.

Reply:
yep, a little unfortunate to have to pass these around, but was quite clear from the types and the example, what needed to be done 👍